stor/dx.py

```diff
@@ -455,29 +687,32 @@ def canonical_resource(self):
         Raises:
             DuplicateError: if filename is not unique
             NotFoundError: if resource is not found on DX platform
+            ValueError : if path is folder path
         """
         if not self.resource:
             return None
-        if self._is_folder():
-            raise ValueError('DXPath ({}) ending in folders cannot be canonicalized'.format(self))
+        if self.endswith('/'):
+            raise ValueError('Cannot call on folder path ({})'.format(self))
         objects = [{
             'name': self.name,
             'folder': self.resource.parent,
-            'project': self.canonical_project
+            'project': self.canonical_project,
+            'batchsize': 2
         }]
         results = dxpy.resolve_data_objects(objects=objects)[0]
         if len(results) > 1:
             raise DuplicateError('The virtual resource is not unique on DNAnexus')
```

Comment:
can this be more clearly phrased? e.g. `Multiple objects found at path %s. Try using a canonical ID instead?`

Reply:
Yes, this could be clearer, agreed. Will change it.